utils/kubernetes/kube_state_processor.py

```diff
@@ -228,38 +286,70 @@ def kube_pod_status_phase(self, message, **kwargs):
-            tags = ['node:{}'.format(self._extract_label_value("node", metric.label))]
-            if name == 'true' and val:
-                self.service_check(service_check_name, self.kube_check.OK, tags=tags)
-            elif name == 'false' and val:
-                self.service_check(service_check_name, self.kube_check.CRITICAL, tags=tags)
-            elif name == 'unknown' and val:
-                self.service_check(service_check_name, self.kube_check.UNKNOWN, tags=tags)
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
 
-    def kube_node_status_out_of_disk(self, message, **kwargs):
-        """ Whether the node is out of disk space. """
-        service_check_name = NAMESPACE + '.node.out_of_disk'
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_requested_cpu_cores(self, message, **kwargs):
+        """ CPU cores requested for a container in a pod. """
+        metric_name = NAMESPACE + '.container.cpu_requested'
         for metric in message.metric:
-            name, val = self._eval_metric_condition(metric)
-            tags = ['node:{}'.format(self._extract_label_value("node", metric.label))]
-            if name == 'true' and val:
-                self.service_check(service_check_name, self.kube_check.CRITICAL, tags=tags)
-            elif name == 'false' and val:
-                self.service_check(service_check_name, self.kube_check.OK, tags=tags)
-            elif name == 'unknown' and val:
-                self.service_check(service_check_name, self.kube_check.UNKNOWN, tags=tags)
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            #TODO: add deployment/replicaset?
+            self.gauge(metric_name, val, tags)
 
-    def kube_node_spec_unschedulable(self, message, **kwargs):
-        """ Whether a node can schedule new pods. """
-        metric_name = NAMESPACE + '.node.status'
-        statuses = ('schedulable', 'unschedulable')
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_requested_memory_bytes(self, message, **kwargs):
+        """ Memory bytes requested for a container in a pod. """
+        metric_name = NAMESPACE + '.container.memory_requested'
         for metric in message.metric:
+            val = metric.gauge.value
             tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
-            status = statuses[int(metric.gauge.value)]  # value can be 0 or 1
-            tags.append('status:{}'.format(status))
-            self.gauge(metric_name, 1, tags)  # metric value is always one, value is on the tags
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_limits_cpu_cores(self, message, **kwargs):
```

Comment:
I see quite a lot of duplication, how about a simple helper function, something like:

```python
def save(self, metric_name, metric):
    for metric in message.metric:
        val = metric.gauge.value
        tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
        self.gauge(metric_name, val, tags)
```

And we could change most of the functions to something like the following:

```python
def kube_pod_container_limits_cpu_cores(self, message, **kwargs):
      """ CPU cores limit for a container in a pod. """
      save(self, NAMESPACE + '.container.cpu_limit', message)
```

Generating code, something I've done for Go and configs but never for Python, could possibly be another way
to make this work a little less tedious and and error prone. But out of scope for a small inline comment in a code review :wink: